numpy/lib/tests/test_shape_base.py

```diff
@@ -713,5 +713,5 @@ def test_basic(self):
 
 # Utility
 def compare_results(res, desired):
-    for i in range(len(desired)):
-        assert_array_equal(res[i], desired[i])
+    for result_item, desired_item in zip(res, desired):
+        assert_array_equal(result_item, desired_item)
```

Comment:
Does this need a if len(res) != len(desired): raise ... or is it guarenteed that they will be equal length in all the tests?

Reply:
I've added a length check, with a comment for [PEP 618](https://www.python.org/dev/peps/pep-0618/) for Python 3.10, when these functions should be refactored to `zip(..., strict=True)`.

It also turns out there was one failure (`test_integer_split_2D_rows`), where the lengths were not equal, which is simple to fix in this PR.